src/mutation.cpp

```diff
@@ -172,6 +171,50 @@ bool Character::has_base_trait( const trait_id &b ) const
     return my_traits.find( b ) != my_traits.end();
 }
 
+int Character::get_instability_per_category( const mutation_category_id &categ ) const
+{
+    int mut_count = 0;
+    bool robust = has_trait( trait_ROBUST );
+    // for each and every trait we have...
+    for( const trait_id &mut : get_mutations() ) {
+        // only count muts that have 0 or more points, aren't a threshold, are valid, and aren't a base trait
+        if( mut.obj().points > -1 && !mut.obj().threshold && mut.obj().valid && !has_base_trait( mut ) ) {
+            bool in_categ = false;
+            // if among all allowed categories the mutation has, the input category is one of them
+            for( const mutation_category_id &Ch_cat : mut.obj().category ) {
+                if( Ch_cat == categ ) {
+                    in_categ = true;
+                }
+            }
+
+            const int height = mutation_height( mut );
+
+            // thus add 1 point if it's in the tree we mutate into, otherwise add 2 points
+            // or if we have Robust Genetics, treat all mutations as in-tree
+            if( in_categ || robust ) {
+                mut_count += height * 1;
+            } else {
+                mut_count += height * 2;
+            }
+        }
+    }
+    return mut_count;
+}
+
+int get_total_nonbad_in_category( const mutation_category_id &categ )
+{
+    int mut_count = 0;
+
+    // iterate through all available traits in this category and count every one that isn't bad or the threshold
```

Comment:
```suggestion
    // For each and every trait we have...
    for( const trait_id &mut : get_mutations() ) {
        // only count muts that have 0 or more points, aren't a threshold, are valid, and aren't a base trait.
        if( mut.obj().points > -1 && !mut.obj().threshold && mut.obj().valid && !has_base_trait( mut ) ) {
            bool in_categ = false;
            // If among all allowed categories the mutation has, the input category is one of them.
            for( const mutation_category_id &Ch_cat : mut.obj().category ) {
                if( Ch_cat == categ ) {
                    in_categ = true;
                }
            }

            const int height = mutation_height( mut );

            // Thus add 1 point if it's in the tree we mutate into, otherwise add 2 points
            // or, if we have Robust Genetics, treat all mutations as in-tree.
            if( in_categ || robust ) {
                mut_count += height * 1;
            } else {
                mut_count += height * 2;
            }
        }
    }
    return mut_count;
}

int get_total_nonbad_in_category( const mutation_category_id &categ )
{
    int mut_count = 0;

    // Iterate through all available traits in this category and count every one that isn't bad or the threshold.
```